Mathlib/CategoryTheory/Monoidal/Braided.lean

```diff
@@ -618,4 +660,108 @@ attribute [reassoc] associator_monoidal
 
 end Tensor
 
+namespace MonoidalOpposite
+
+instance instBraiding : BraidedCategory Cᴹᵒᵖ where
+  braiding X Y := (β_ (unmop Y) (unmop X)).mop
+  braiding_naturality_right X {_ _} f := braiding_naturality_left f.unmop (unmop X)
+  braiding_naturality_left {_ _} f Z := braiding_naturality_right (unmop Z) f.unmop
+  hexagon_forward X Y Z := hexagon_reverse (unmop Z) (unmop Y) (unmop X)
+  hexagon_reverse X Y Z := hexagon_forward (unmop Z) (unmop Y) (unmop X)
+
+/-- The identity functor on `C`, viewed as a functor from `C` to its
+monoidal opposite, upgraded to a braided functor. -/
+@[simps!] def mopBraidedFunctor : BraidedFunctor C Cᴹᵒᵖ where
+  μ X Y := (β_ (mop X) (mop Y)).hom
+  ε := 𝟙 (𝟙_ Cᴹᵒᵖ)
+  associativity X Y Z := by
+    simp only [tensorHom_id, id_tensorHom]
+    change (β_ (mop X) (mop Y)).hom ▷ (mop Z) ≫ (β_ (mop Y ⊗ mop X) (mop Z)).hom ≫
+              (α_ (mop Z) (mop Y) (mop X)).inv
+            = (α_ (mop X) (mop Y) (mop Z)).hom ≫ (mop X) ◁ (β_ (mop Y) (mop Z)).hom ≫
+                (β_ (mop X) (mop Z ⊗ mop Y)).hom
+    refine (α_ (mop X) (mop Y) (mop Z)).inv_comp_eq.mp ?_
+    simp only [← assoc]
+    refine (α_ (mop Z) (mop Y) (mop X)).comp_inv_eq.mpr ?_
+    simp only [braiding_tensor_left, braiding_tensor_right,
+               assoc, Iso.inv_hom_id, comp_id]
+    exact yang_baxter (mop X) (mop Y) (mop Z)
+  left_unitality X :=
+    Eq.symm $ Eq.trans (congrArg (. ≫ _) (tensor_id _ _))
+    $ Eq.trans (id_comp _) (braiding_rightUnitor Cᴹᵒᵖ (mop X))
```

Comment:
In this proof (and also similar proofs below), it would be advisable to use sequences of `rw` (or maybe `erw` if is it necessary), and probably use `_assoc` versions of some lemmas.

Reply:
Oh I didn't realize erw would work, nice! I haven't really used that tactic before